core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -48,6 +48,27 @@ private CatalogProperties() {}
 
   public static final boolean CACHE_CASE_SENSITIVE_DEFAULT = true;
 
+  /**
+   * Controls policy of caffeine cache
+   *
+   * <p>Behavior of specific values of cache.strategy:
+   *
+   * <ul>
+   *   <li>EXPIRE_AFTER_ACCESS - cache entries are never evicted as long as they are being accessed
+   *       frequently
+   *   <li>EXPIRE_AFTER_WRITE - cache entries are evicted frequently after cache write
+   * </ul>
+   */
+  public static final String CACHE_EXPIRATION_POLICY = "cache.expiration-policy";
+
+  public enum CacheExpirationPolicy {
```

Comment:
Could we have one which doesn't have an expiration like before?

Reply:
I'm afraid, that having an expiration is a behavioral change which could cause issues for some users

Reply:
Default cache expiration is exactly like before, so users would not impact by this change